### PR TITLE
Update squeeze op to allow non-one axes

### PR DESCRIPTION
**Pull request by daquexian**
Fix issue https://github.com/onnx/onnx/issues/2466

### CODE REVIEW

File: onnx/defs/tensor/defs.cc [hunk 1]
```diff
@@ -1237,7 +1237,7 @@ ONNX_OPERATOR_SET_SCHEMA(
           }
         }));
 
-static const char* Squeeze_ver11_doc = R"DOC(
+static const char* Squeeze_ver12_doc = R"DOC(
 Remove single-dimensional entries from the shape of a tensor.
```
Comment (linkerzhang) [inline onnx/defs/tensor/defs.cc]: this definition should be updated, given non-one axis value is supported (with no change to the input).

**Comment by linkerzhang** [comment]
@daquexian do you mind to update this PR with resolving the comments and conflicts please? thanks!

**Comment by daquexian** [comment]
@linkerzhang Sure, I'll update it soon

**Comment by daquexian** [comment]
I'll update the docs tomorrow

**Comment by daquexian** [comment]
@linkerzhang The CI error is not related to this PR. Is it ok to merge it?

File: onnx/defs/tensor/defs.cc [hunk 2]
```diff
@@ -1331,23 +1332,18 @@ ONNX_OPERATOR_SET_SCHEMA(
...
+            if (std::find(axes.begin(), axes.end(), i) != axes.end() &&
+                input_shape.dim(i).has_dim_value() &&
+                input_shape.dim(i).dim_value() == 1) {
+              continue;
             }
+            *ctx.getOutputType(0)
+                 ->mutable_tensor_type()
+                 ->mutable_shape()
+                 ->add_dim() = input_shape.dim(i);
           }
```
Comment (TMVector) [inline onnx/defs/tensor/defs.cc]: This will infer an incorrect rank!

**Comment by daquexian** [comment]
Closing this PR as I think @TMVector is right :D

**Pull request closed without merging**